tools/ob_admin/backup_tool/ob_admin_backup_validation_util.cpp

```diff
@@ -362,5 +362,124 @@ int ObAdminBackupValidationUtil::read_macro_block_data(
   }
   return ret;
 }
+
+int ObAdminBackupValidationUtil::get_ls_ids_from_active_piece(
+    const share::ObBackupDest &backup_piece_dest, common::ObArray<share::ObLSID> &ls_ids,
+    ObAdminBackupValidationCtx *ctx)
+{
+  int ret = OB_SUCCESS;
+  ObBackupIoAdapter util;
+  ObArray<ObIODirentEntry> d_entrys;
+  ObDirPrefixEntryNameFilter prefix_op(d_entrys);
+  char logstream_prefix[OB_BACKUP_DIR_PREFIX_LENGTH] = {0};
+  if (OB_FAIL(databuff_printf(logstream_prefix, sizeof(logstream_prefix), "%s", "logstream_"))) {
+    STORAGE_LOG(WARN, "failed to set log stream prefix", K(ret));
+  } else if (OB_FAIL(prefix_op.init(logstream_prefix, strlen(logstream_prefix)))) {
+    STORAGE_LOG(WARN, "failed to init dir prefix", K(ret), K(logstream_prefix));
+  } else if (OB_FAIL(util.list_directories(backup_piece_dest.get_root_path(),
+                                           backup_piece_dest.get_storage_info(), prefix_op))) {
+    STORAGE_LOG(WARN, "failed to list files", K(ret));
+  } else {
+    ObIODirentEntry tmp_entry;
+    ObLSID ls_id;
+    for (int64_t i = 0; OB_SUCC(ret) && i < d_entrys.count(); ++i) {
+      int64_t id_val = 0;
+      tmp_entry = d_entrys.at(i);
+      if (OB_ISNULL(tmp_entry.name_)) {
+        ret = OB_ERR_UNEXPECTED;
+        STORAGE_LOG(WARN, "file name is null", K(ret));
+      } else if (sscanf(tmp_entry.name_, "logstream_%ld", &id_val) != 1) {
+        ret = OB_ERR_UNEXPECTED;
+        STORAGE_LOG(WARN, "file name is null", K(ret));
+      } else if (FALSE_IT(ls_id = id_val)) {
+      } else if (OB_FAIL(ls_ids.push_back(ls_id))) {
+        STORAGE_LOG(WARN, "failed to push back ls_ids", K(ret));
+      }
+    }
+  }
+  if (OB_NOT_NULL(ctx) && OB_FAIL(ret)) {
+    ctx->go_abort(backup_piece_dest.get_root_path().ptr(), "File not exist or physical corrputed");
+  }
+  return ret;
+}
+
+int ObAdminBackupValidationUtil::get_piece_ls_start_lsn(
+    const share::ObBackupDest &backup_piece_dest, const share::ObLSID &ls_id, uint64_t &start_lsn,
+    ObAdminBackupValidationCtx *ctx)
+{
+  int ret = OB_SUCCESS;
+  class ObLSStartFileGetterOp final : public ObBaseDirEntryOperator
```

Comment:
prefer not to define complicated class in method

Reply:
fixed